docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,370 @@
+<!--MarkDown doesn't support embedded YouTube players, which is one reason I removed the videos. Another is that it's better practice to link to official sites, not to videos uploaded by "unofficial" users who may remove them without warning and may, for all we know, be in legal disputes with the companies who created those products (I admit that isn't very likely with "how to install" videos). Also, not sure these videos will work in all countries.-->
+### Download, build and install the application
 
-![IMAGE ALT TEXT HERE](https://raw.githubusercontent.com/COTASPAR/K66F/master/images/nxp_k66f_board.jpeg)
+1. Navigate to the directory where you keep code projects.
+1. Download the TensorFlow Lite source code:
 
+    ```
+    git clone https://github.com/tensorflow/tensorflow.git
+    ```
 
-We will deploy a sample application that uses the microphone on the K66F and a TensorFlow machine learning model to detect the words “yes” and “no”.
+1. Navigate into the project directory:
 
-To do this, we will show you how to complete the following steps:
+    ```
+    cd tensorflow/tensorflow/lite/micro/mbed/
+    ```
+    
+1. Change the DebugLog function to use UnbufferedSerial instead since the Serial API was deprecated in Mbed OS 6:
+    ```
+    nano debug_log.cc
+    
+    extern "C" void DebugLog(const char* s) {
+    UnbufferedSerial pc(USBTX, USBRX,9600);
+    FILE *out = fdopen(&pc,"w");
+    fprintf(out, s);
+     }
+    ```
+1. Build the project
 
-1.  Download and build the sample application
+    ```
+    cd ../../../../
 
+    make -f tensorflow/lite/micro/tools/make/Makefile TARGET=mbed TAGS="nxp_k66f" generate_micro_speech_mbed_project
+    ```
+    <!--I got a "no such file or directory error". I think it's because there was already `cd tensorflow` and then the path in the command also has `tensorflow`. Removing `tensorflow` from the path removed the error-->
 
-2.  Deploy the sample to your K66F
+    This creates a folder in `tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed` that contains the source and header files for the example application, Mbed OS driver files, and a README.
 
+1. Navigate into the `mbed` folder:
 
-3.  Use new trained models to recognize different words
+    ```
+    cd tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed
+    ```
+1. Connect your board to your computer over USB. On your board, when the Ethernet port is facing you, the micro USB port is to its left.
+    Your board appears as storage on your computer. If your system does not recognize the board with the `mbed detect` command, follow the instructions for setting up [DAPLink](https://armmbed.github.io/DAPLink/?board=FRDM-K66F).
+    
+1. Execute the following to flash the application onto the board:
 
-# Download and build the sample application
+    ```
+    mbed config root .
```

Comment:
Probably a shortcoming of the TFLiteU Makefile above, ideally this wouldn't be needed so candidate for upstreaming too.